apps/pie-storybook/stories/pie-button.stories.ts

```diff
@@ -3,12 +3,12 @@ import type { StoryObj as Story } from '@storybook/web-components';
 import {
     ButtonProps as ButtonPropsBase, sizes, types, variants,
 } from '@justeattakeaway/pie-button';
+import { IconChevronDown, IconPlusCircle } from '@justeattakeaway/pie-icons-webc';
 import { StoryMeta, SlottedComponentProps } from '../types';
 
-import '@justeattakeaway/pie-icons-webc/icons/IconPlusCircle'; // Register icon-plus-circle
-import '@justeattakeaway/pie-icons-webc/icons/IconChevronDown'; // Register icon-chevron-down
+const keptReferences = [IconChevronDown, IconPlusCircle];
 
-type ButtonProps = SlottedComponentProps<ButtonPropsBase>;
+type ButtonProps = SlottedComponentProps<ButtonPropsBase> & { buttonIcons : Array<string> };
```

Comment:
Sure. Without this, the linter complains that the `defaultArgs` we provide don't match the properties of `ButtonProps`. This line extends that type by adding a new property for the `buttonIcons` array that you added.

Reply:
📓 no action is required for the following but more of a knowledge sharing.

the following suggestion will probably create the same thing but considering `SlottedComponentProps` is generic that should wrap _all_ types, I'd have extended the generic value instead of the whole type.

```suggestion
type ButtonProps = SlottedComponentProps<ButtonPropsBase & { buttonIcons : Array<string> }>;
```

also at this point, I'm not sure about the value of having `SlottedComponentProps` when the following would be a lot easier to customize.

```suggestion
type ButtonProps = ButtonPropsBase & { slot: string; buttonIcons : Array<string> };
```

we shouldn't keep adding generics for each property we want to add as this is usually done through extending interfaces from a base interface or with `&`. generics are useful when we want to handle multiple dynamic types as they provide flexibility of types at compile time.  for example:

```
const fetchData<T> = (data) => {
const res: {
isLoading: boolean;
data: T } =  {
  isLoading: false,
  data: data
}
return res
}

fetchData<string[]>(['Hellow'])
```

Reply:
I think `SlottedComponentProps` is still valuable because there will be a lot of components that take a slot, however only this component (and probably the form input?) will take a list of icons so it didn't make much sense to me to pull that out into another generic wrapper.

In terms of where to put the `>`, I'm not sure it makes too much difference but personally I prefer the way the first one reads